metal/bgp.go

```diff
@@ -47,35 +51,42 @@ func (b *bgp) enableBGP() error {
 	// - bgpConfig struct exists
 	// - bgpConfig struct has non-blank ID
 	// - bgpConfig struct does not have Status=="disabled"
-	if err == nil && bgpConfig != nil && bgpConfig.ID != "" && strings.ToLower(bgpConfig.Status) != "disabled" {
-		b.localASN = bgpConfig.Asn
-		b.bgpPass = bgpConfig.Md5
+	if err == nil && bgpConfig != nil && bgpConfig.GetId() != "" && bgpConfig.GetStatus() != metal.BGPCONFIGSTATUS_DISABLED {
+		b.localASN = int(bgpConfig.GetAsn())
+		b.bgpPass = *bgpConfig.Md5.Get()
 		return nil
 	}
 
 	// we did not have a valid one, so create it
-	req := packngo.CreateBGPConfigRequest{
-		Asn:            b.localASN,
-		Md5:            b.bgpPass,
+	req := metal.BgpConfigRequestInput{
+		Asn:            int32(b.localASN),
+		Md5:            &b.bgpPass,
 		DeploymentType: "local",
-		UseCase:        "kubernetes-load-balancer",
+		UseCase:        pointer.String("kubernetes-load-balancer"),
```

Comment:
the `metalv1` package has helpers for this so you can avoid pulling in an extra dependency; since you've aliased the package as `metal` here, the call would be [`metal.PtrString("kubernetes-load-balancer")`](https://github.com/equinix/equinix-sdk-go/blob/main/services/metalv1/utils.go#L39)

Reply:
Ah right. Didnt know. Changed